sensors/triggers/params.go

```diff
@@ -160,8 +180,9 @@ func renderEventDataAsJSON(event *v1alpha1.Event) ([]byte, error) {
 }
 
 // helper method to resolve the parameter's value from the src
+// returns value and value type (jsonType or stringType or errorType). jsonType represent a block while stringType represent a single value.
 // returns an error if the Path is invalid/not found and the default value is nil OR if the eventDependency event doesn't exist and default value is nil
-func ResolveParamValue(src *v1alpha1.TriggerParameterSource, events map[string]*v1alpha1.Event) (*string, error) {
+func ResolveParamValue(src *v1alpha1.TriggerParameterSource, events map[string]*v1alpha1.Event) (*string, string, error) {
```

Comment:
Even you added an `errorType` in the return, but you never checked it. To make it simple, the guideline should be:

1. If there's no error, return `(xxx, xx, nil)`
2. Or return `(nil, "", err)`

Reply:
that what originally been there...  reverting those changes..

Reply:
Done